### ISSUE TITLE
[native CuteDSL FFA] Long Term native CuteDSL FFA dev Progress Tracker

### ISSUE THREAD

**Issue by cherhh** [SandAI-org/MagiAttention#349]
To track native CuteDSL FFA dev progress(Implement the FFA feature based on the native CuteDSL codebase(#327 )).
## DONE

- **Native ranges + mask types in-kernel**: let the kernel perceive q_ranges / k_ranges + mask_types directly, instead of the hacky cu_seqlens lowering.

## TODO

- **Dtype & determinism**: configurable fwd/bwd output dtype; fwd/bwd deterministic mode.
- **GQA:** pack-gqa / cat-gqa support.
- **Attn sink**: learnable attn-sink in fwd post-process / bwd pre-process.
- **Muon**: fwd Muon qk-clip max-logits.
- **sparse attention support**

#### basic qk range

- #344
